docs/website/blog/2024-01-11-dlt-dbt-semantic-layer.md

```diff
@@ -0,0 +1,135 @@
+---
+slug: dlt-dbt-semantic-layer
+title: "dlt & dbt in Semantic Modelling"
+image: /img/blog-dbt_sem-data-people.png
+authors:
+  name: Hiba Jamal
+  title: Data Science intern at dlthub
+  url: https://github.com/hibajamal
+  image_url: https://avatars.githubusercontent.com/u/35984866?v=4
+tags: [semantic modelling]
+---
+
+<aside>
+💡 TLDR; Thanks to dbt’s new semantic layer feature we can now write “data pipeline” and “business user access” in the same sentence. In this article we explore how to create an end to end pipeline using the dlt library and dbt’s new feature.
+</aside>
+
+## The Chinese Whisper of Data
+
+In the context of constructing a **modern data stack** through the development of various modular components for a data pipeline, our attention turns to the centralization of metrics and their definitions.
+
+For the purposes of this demo, we’ll be looking specifically at how dlt and dbt come together to solve the problem of the data flow from data engineer → analytics engineer → data analyst → business user. That’s quite a journey. And just like any game of *Chinese whisper*, things certainly do get lost in translation.
+
+<div style={{ paddingRight: '10%', paddingLeft: '10%', paddingBottom: '1%' }}>
+
+![cover](/img/blog-dbt_sem-data-people.png)
+<div>Taken from the real or fictitious book called '5th grade data engineering, 1998'.</div>
+
+</div>
+
+To solve for this problem, both these tools come together and seamlessly integrate to create everything from data sources to uniform metric definitions, that can be handled centrally, and hence are a big aid to the data democracy practices of your company!
+
+Here’s how a pipeline could look:
+1. Extract and load with dlt: Dlt will automate data cleaning and normalization leaving you with clean data you can just use.
+2. Create SQL models that simplify sources, if needed. This can include renaming and/or eliminating columns, identifying and setting down key constraints, fixing data types, etc.
+3. Create and manage central metric definitions with the semantic layer.
+
+## 1. Extract, Structure, & Load with dlt
+
+The data being used is of a questionnaire, which includes questions, the options of those questions, respondents and responses. This data is contained within a nested json object, that we’ll pass as a raw source to dlt to structure, normalize and dump into a BigQuery destination.
+
+```python
+# initializing the dlt pipeline with your data warehouse destination
+pipeline = dlt.pipeline(
+        pipeline_name="survey_pipeline",
+        destination="bigquery",
+        dataset_name="questionnaire")
```

Comment:
I'm sorry for nitpicking about little things. we use black code style, it means line breaking should be in black style even in the blog
```
pipeline = dlt.pipeline(
    pipeline_name="survey_pipeline",
    destination="bigquery",
    dataset_name="questionnaire"
)
```

Reply:
and only 4 spaces here
`    pipeline_name="survey_pipeline",`

Reply:
aah I was unaware of that - thank you!